File: 0926-flip-string-to-monotone-increasing/0926-flip-string-to-monotone-increasing.js
/**
 * @param {string} s
 * @return {number}
 */
var minFlipsMonoIncr = function(s) {
    let ans = 0, n1 = 0;
    for (let i of s)
        if (i === '0') ans = Math.min(n1, ans + 1);
        else ++n1;
    return ans;
};

File: 0926-flip-string-to-monotone-increasing/0926-flip-string-to-monotone-increasing.cpp
class Solution {
public:
    int minFlipsMonoIncr(string s) {
        int ans = 0, n1 = 0;
        for (auto &c : s)
            if (c == '0') ans = min(n1, ans + 1);
            else ++n1;
        return ans;
    }
};